File: simple_client.js
const TUNNEL_URL = 'https://user225847803-f3ep5hdc.wormhole.vk-apps.com/'  // url, который дал вам туннель
// const TUNNEL_URL = 'wss://127.0.0.1:3002/'  // url, который дал вам туннель
// const TUNNEL_URL = 'ws://127.0.0.1:3000/'  // url, который дал вам туннель

const io = require('socket.io-client')

const socket = io(TUNNEL_URL, {
    transports: ['websocket'],
    rejectUnauthorized: false,
    // extraHeaders: {
    //     Host: "https://user225847803-f3ep5hdc.wormhole.vk-apps.com/"
    // }
})

socket.on('connect', () => {
  console.log('Connected to server')
  setInterval(() => socket.emit('message', 'Hello from client'), 2000)
})

socket.on('message', (msg) => {
  console.log('Message from server:', msg)
})

socket.on('disconnect', () => {
  console.log('Disconnected from server')
})

File: 002/services/detail/simple_client.py
import asyncio
import socketio

sio = socketio.AsyncClient(ssl_verify=False)

@sio.event
async def connect():
    print('connection established')

@sio.event
async def message(data):
    print('message received with ', data)

@sio.event
async def disconnect():
    print('disconnected from server')

async def sender():
    while 1:
        await asyncio.sleep(1)
        await sio.send('123')

async def main():
    await sio.connect('https://user225847803-yybmoqi2.wormhole.vk-apps.com/', transports=['websocket'])
    asyncio.create_task(sender())
    await sio.wait()

if __name__ == '__main__':
    asyncio.run(main())
